packages/frontend/src/components/ChooseConnectionDropdown/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { IApp } from '@plumber/types'
+
+import { useCallback, useState } from 'react'
+import { FormControl } from '@chakra-ui/react'
+import { BxPlus, FormLabel, SingleSelect } from '@opengovsg/design-system-react'
+import AddAppConnection from 'components/AddAppConnection'
+
+type ConnectionDropdownOption = {
+  label: string
+  value: string
+  icon?: React.ElementType
+}
+interface ChooseConnectionDropdownProps {
+  isDisabled: boolean
+  connectionOptions: ConnectionDropdownOption[]
+  onChange: (value: string, shouldRefetch: boolean) => void
+  value: string | undefined
+  application: IApp
+}
+
+const ADD_CONNECTION_VALUE = 'ADD_CONNECTION'
+const ADD_NEW_CONNECTION_OPTION: ConnectionDropdownOption = {
+  label: 'Add new connection',
+  icon: BxPlus,
+  value: ADD_CONNECTION_VALUE,
+}
+
+function ChooseConnectionDropdown({
+  isDisabled,
+  connectionOptions,
+  onChange,
+  value,
+  application,
+}: ChooseConnectionDropdownProps) {
+  const [showAddConnectionDialog, setShowAddConnectionDialog] = useState(false)
```

Comment:
nit: how about using `useDisclosure`

Reply:
honestly, im not using the other properties of useDisclosure since the onClose is handed by the Dialog component. Will refactor when rewriting `AddAppConnection` component